app/views.py

```diff
@@ -0,0 +1,102 @@
+from flask import current_app, Blueprint, render_template, abort, request, url_for, flash, redirect
+from jinja2 import TemplateNotFound
+from .forms import AddForm
+from sqlalchemy import create_engine, Column, Integer, String, func
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+main_page = Blueprint('mainpage ', __name__)
+
+# init database
+engine = create_engine('sqlite:///books.db', echo=False)
+Base = declarative_base()
+Session = sessionmaker(bind=engine)
+session = Session()
+
+class Book(Base):
+    __tablename__ = 'books'
+
+    id = Column(Integer, primary_key=True)
+    name = Column(String)
+    author = Column(String)
+    pages = Column(Integer)
+
+    def __repr__(self):
+        return "<Book(name='%s', author='%s', pages='%s')>" % (
+                            self.name, self.author, self.pages)
+
+Base.metadata.create_all(engine)
+
+def update_books(book, action=''):
+    if action == 'delete':
+        session.delete(book)
+    elif action == 'add':
+        session.add(book)
+    session.commit()
+
+@main_page.route('/<page>')
+def show(page):
+    try:
+        return render_template('%s.html' % page)
+    except TemplateNotFound:
+        abort(404)
+
+@main_page.route('/')
+@main_page.route('/index')
+@main_page.route('/list')
+def list():
+    total_rows = session.query(func.count(Book.name)).scalar()
+    books = session.query(Book).order_by(Book.id)
+
+    return render_template("list.html",
+        title = 'Current list (' + str(total_rows) + ')',
+         books = books)
+
+@main_page.route('/add', methods = ['GET', 'POST'])
+def add():
+    form = AddForm(request.form)
+    if request.method == 'GET':
+        return render_template('add.html',
+            title = 'Adding new book',
+            form = form)
+    elif request.method == 'POST':
+        if form.validate_on_submit():
+            book = Book(name=request.form['book_name'], author='', pages=0)
+            flash('Book "' + request.form['book_name'] + '" added!')
+            update_books(book, 'add')            
+            return redirect('/')
+        else:
+            flash('Adding failed! Fill all fields!')
+            return redirect('/add')
+
+@main_page.route('/edit/<index>', methods = ['GET', 'POST'])
+def edit(index):
+    form = AddForm(request.form)
+    book = session.query(Book).filter(Book.id == int(index)).all()
+    form.book_name.data = book[0].name
+    form.book_author.data = book[0].author
+    form.book_pages.data = book[0].pages
+
+    if request.method == 'GET':
+        return render_template('add.html',
+            title = 'Editing book',
+            form = form)
+    elif request.method == 'POST':
+        if form.validate_on_submit():
+            book = session.query(Book).filter(Book.id == int(index)).all()
+            book[0].name = request.form['book_name']

```

Comment:
WTForms have possibility to populate the object. You don't have to assign things one by one. See docs